code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -135,6 +135,22 @@ var/const/VOX_PATH = "sound/vox_fem/"
 	for(var/word in words)
 		play_vox_word(word, src.z, null)
 
+	ai_voice_announcement_to_text(words, src)
+
+
+/proc/ai_voice_announcement_to_text(words, ai)

```

Comment:
Optional (might be out of the scope of the PR):
You can probably use the `datum/announcement` object for this. It'll handle all things in here except for the z level thing. 
However it would require (quite) some refactoring of that system.

Reply:
Yeeeep. I remember trying to use that I think. I don't want to touch that right now :smile: